chia/data_layer/data_layer.py

```diff
@@ -112,6 +112,12 @@ async def batch_update(
         changelist: List[Dict[str, Any]],
         fee: uint64,
     ) -> TransactionRecord:
+
+        # check before any DL changes that this singleton is currently owned by this wallet
```

Comment:
I'm not sure if this checks if we're _subscribed_ to the singleton, or if we _own_ it. If we want owned only singletons, this _might_ give us subscribed singletons as well (I use the same method in `fetch_and_validate`, which is for subscriptions, not only for owned singletons).

Reply:
I believe it will give subscribed singletons as well - but I was trying to catch the most obvious case with the simplest check - I suppose I could use the new get_owned call to return the list of owned singletons. Maybe that is better